lib/assetmanager.js

```diff
@@ -215,7 +216,37 @@
         var layerIdsWithComponents = [],
             compComponents = [],
             doc = this._document,
-            comps = doc._comps;
+            comps = doc._comps,
+            docMeta = this._document._generatorSettings[META_PLUGIN_ID];
+        
+        if (docMeta && docMeta.json) {
+            docMeta = JSON.parse(docMeta.json);
```

Comment:
wrap this in try/catch. JSON.parse will throw on any syntax error. Especially for anything coming from an outside source you got to be careful.